pkg/cmd/cli/cmd/rollback.go

```diff
@@ -77,31 +80,60 @@ func NewCmdRollback(fullName string, f *clientcmd.Factory, out io.Writer) *cobra
 			cmdutil.CheckErr(err)
 			oClient, kClient, err := f.Clients()
 			cmdutil.CheckErr(err)
+			helper := newHelper(oClient, kClient)
 
-			// Set up the rollback config
-			rollback.Spec.From.Name = args[0]
+			// Get the deploymentConfig resource referenced in the command args.
+			// TODO: The resource.Builder should handle the prefix defaulting
+			// automatically, but won't until [1] is closed.
+			// [1] https://github.com/GoogleCloudPlatform/kubernetes/pull/10979
+			configName := args[0]
+			if strings.Index(configName, "/") == -1 {
+				configName = "dc/" + configName
+			}
+			mapper, typer := f.Object()
+			r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+				NamespaceParam(namespace).
+				ResourceTypeOrNameArgs(false, configName).
+				SingleResourceType().
+				Do()
+			cmdutil.CheckErr(r.Err())
+			obj, err := r.Object()
+			cmdutil.CheckErr(err)
+			config, ok := obj.(*deployapi.DeploymentConfig)
+			if !ok {
+				cmdutil.CheckErr(fmt.Errorf("%s is not a valid deploymentconfig", configName))
+			}
 
-			// Make a helper and generate a rolled back config
-			helper := newHelper(oClient, kClient)
-			config, err := helper.Generate(namespace, rollback)
+			// Find the target deployment by the specified version, or by a lookup
+			// of the last completed deployment.
+			desiredVersion := 0
+			if cmd.Flag("to-version").Changed {
```

Comment:
couldn't you just default `to-version` to `0` and then say that `to-version=0` means auto-detect?